daft/sql/sql_scan.py

```diff
@@ -272,7 +269,6 @@ def _construct_scan_task(
             file_format=file_format_config,
             schema=self._schema._schema,
             num_rows=num_rows,
-            storage_config=self.storage_config,
```

Comment:
Why is this no longer required?

Reply:
the storage config was only passed in as a dummy one to stick into the required storage config field in a scan task. read_sql doesn't actually use the storage config in any way, so I just removed it from everywhere in the call graph until the actual creation of the scan task in the Rust side

Reply:
Reverted, see comment in src/daft-scan/src/python.rs